### PR TITLE
resolves issue with media url return data

### PR DESCRIPTION
**Pull request by AliceYuan**
adds new api_query_url field for api queries
added try catch for checking for api response
youtube now does both api and scraping of header for content
resolves issue: #71

### CODE REVIEW

**Comment by samiyaakhtar** [comment]
r+

File: client/scraper/youtube_metadata.py [hunk 1]
```diff
@@ -11,33 +15,51 @@
...
+        if api_response.status_code is StatusCode.OK:
+            response.set_api_content(api_response.content)
+            response.status_code = api_response.status_code
+            response.api_url = api_url
+
+        return response
 
     def parse_content(self, response):
         """
         The youtube metadata parse_content does not call generic_parse_content, since the response is a JSON object
```
Comment (MeenuV) [inline client/scraper/youtube_metadata.py]: Remove this comment?

File: client/scraper/youtube_metadata.py [hunk 2]
```diff
@@ -11,33 +15,51 @@
...
+                        FieldKeyword.IFRAME: data["html"]
+                    }]
+                images_list[FieldKeyword.COUNT] = len(images_list[FieldKeyword.DATA])
+                self.prop_map[FieldKeyword.IMAGES] = images_list
+                self.prop_map[FieldKeyword.API_QUERY_URL] = response.api_url
+
+                media_list[FieldKeyword.COUNT] = len(media_list[FieldKeyword.DATA])
+                self.prop_map[FieldKeyword.MEDIA] = media_list
+            except Exception:
+                # exception in reading json response
```
Comment (MeenuV) [inline client/scraper/youtube_metadata.py]: Maybe write a TODO for logging this error.

File: client/constants.py [hunk 3]
```diff
@@ -43,22 +43,23 @@ def get_special_url(url_type):
 
 
 class FieldKeyword(object):
+    API_QUERY_URL = "api_query_url"
```
Comment (MeenuV) [inline client/constants.py]: Maybe write a comment to say exactly what this field denotes?

**Comment by MeenuV** [comment]
+1 with very minor changes

File: client/scraper/youtube_metadata.py (hunk 1 above)
Comment (AliceYuan) [inline client/scraper/youtube_metadata.py]: ?? why

Comment (AliceYuan) [inline client/scraper/youtube_metadata.py]: jokes nvm

Comment (MeenuV) [inline client/scraper/youtube_metadata.py]: lol, I probably should have cited the line number where we use it

**Pull request merged**